pkg/database/object_database.go

```diff
@@ -0,0 +1,356 @@
+		value := &structpb.Value{}
+		switch tv.Type() {
+		case sql.VarcharType:
+			value.Kind = &structpb.Value_StringValue{StringValue: tv.Value().(string)}
+			return value, nil
+		case sql.IntegerType:
+			value.Kind = &structpb.Value_NumberValue{NumberValue: float64(tv.Value().(int64))}
+			return value, nil
+		case sql.BLOBType:
+			value.Kind = &structpb.Value_StringValue{StringValue: string(tv.Value().([]byte))}
+			return value, nil
+		}
+
+		return nil, errType
+	}
+)
+
+// GetCollection returns the collection schema
+func (d *db) GetCollection(ctx context.Context, req *schemav2.CollectionGetRequest) (resp *schemav2.CollectionInformation, err error) {
+	tx, err := d.objectEngine.NewTx(ctx, sql.DefaultTxOptions().WithReadOnly(true))
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Cancel()
+
+	// check if collection exists
+	table, err := tx.Catalog().GetTableByName(d.name, req.Name)
+	if err != nil {
+		return nil, fmt.Errorf("collection %s does not exist", req.Name)
+	}
+
+	// fetch primary and index keys from collection schema
+	indexes := table.GetIndexes()
+	if len(indexes) == 0 {
+		return nil, fmt.Errorf("collection %s does not have a indexes", req.Name)
+	}
+
+	// iterate over indexes and extract primary and index keys
+	for _, idx := range indexes {
+		for _, col := range idx.Cols() {
+			var colType schemav2.PossibleIndexType
+			switch col.Type() {
+			case sql.VarcharType:
+				colType = schemav2.PossibleIndexType_STRING
+			case sql.IntegerType:
+				colType = schemav2.PossibleIndexType_INTEGER
+			case sql.BLOBType:
+				colType = schemav2.PossibleIndexType_STRING
+			}
+
+			// check if primary key
+			if idx.IsPrimary() {
+				resp.PrimaryKeys[col.Name()] = colType
+			} else {
+				resp.IndexKeys[col.Name()] = colType
+			}
+		}
+	}
+
+	return resp, nil
+}
+
+// CreateCollection creates a new collection
+func (d *db) CreateCollection(ctx context.Context, req *schemav2.CollectionCreateRequest) error {
+	collectionName := req.Name
+	primaryKeys := make([]string, 0)
+	collectionSpec := make([]*sql.ColSpec, 0, len(req.PrimaryKeys))
+
+	// validate collection to contain at least one primary key
+	if len(req.PrimaryKeys) == 0 {
+		return fmt.Errorf("collection must have at least one primary key")
+	}
+
+	// validate primary keys
+	for name, pk := range req.PrimaryKeys {
+		schType, isValid := schemaToValueType[pk]
+		if !isValid {
+			return fmt.Errorf("invalid primary key type: %v", pk)
+		}
+		primaryKeys = append(primaryKeys, name)
+		// TODO: add support for other types
+		// TODO: add support for max length
+		// TODO: add support for auto increment
+		collectionSpec = append(collectionSpec, sql.NewColSpec(name, schType, 0, false, false))
+	}
+
+	// add support for blob
+	// TODO: add support for max length for blob storage
+	collectionSpec = append(collectionSpec, sql.NewColSpec(defaultObjectBLOBField, sql.BLOBType, 10000, false, false))
+
+	_, _, err := d.objectEngine.ExecPreparedStmts(
```

Comment:
concrete implementation should be inside the objectEngine

Reply:
moved